flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/codegen/calls/ScalarFunctions.scala

```diff
@@ -49,6 +49,11 @@ object ScalarFunctions {
     STRING_TYPE_INFO,
     BuiltInMethod.SUBSTRING.method)
 
+  addSqlFunctionTrim(
```

Comment:
Can you briefly explain why `TRIM` cannot be handled as a generic SQL method like `SUBSTRING` and requires its own `TrimCallGenerator`?